### PR TITLE
fix: improve cache logic to handle BigInt

### PR DESCRIPTION
**Pull request by alan-agius4**
Closes: #2375

### CODE REVIEW

File: src/lib/utils/cache.ts [hunk 1]
```diff
@@ -18,22 +18,56 @@ export async function generateKey(...valuesToConsider: string[]): Promise<string
...
 
-export async function saveCacheEntry(cachePath: string, key: string, content: string): Promise<void> {
-  await cacache.put(cachePath, key, content);
+export async function saveCacheEntry(cachePath: string, key: string, content: any): Promise<void> {
+  const filePath = join(cachePath, key);
+
+  // Ensure the cache directory exists
+  await ensureCacheDirExists(cachePath);
+
+  const data = JSON.stringify(content, (_key, value) => (typeof value === 'bigint' ? value.toString() + 'n' : value));
```
Comment (JoostK) [inline src/lib/utils/cache.ts:70]: WDYT of making this a bit more explicit, like `%BigInt(${value})` to reduce the chance of stripping off an `n` suffix of a string that happens to be `(\d+)n`

Comment (alan-agius4) [reply src/lib/utils/cache.ts:70]: Good idea.

File: src/lib/utils/cache.ts [hunk 2]
```diff
@@ -11,29 +11,65 @@ try {
   ngPackagrVersion = require('../../../package.json').version;
 }
 
+const BIGINT_STRING_VALUE_REGEXP = /%BigInt\((\d+)\)/;
```
Comment (JoostK) [inline src/lib/utils/cache.ts:14]:
```suggestion
const BIGINT_STRING_VALUE_REGEXP = /^%BigInt\((\d+)\)$/;
```

**Review by JoostK — COMMENTED** [review]
I think this should work, just one suggestion to anchor the regex to reduce its search space.

**Review by JoostK — APPROVED** [review]

**Pull request merged**